impl/src/main/kotlin/nl/nuts/consent/bridge/listener/StateFileStorage.kt

```diff
@@ -0,0 +1,155 @@
+/*
+ * Nuts consent bridge
+ * Copyright (C) 2020 Nuts community
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package nl.nuts.consent.bridge.listener
+
+import nl.nuts.consent.bridge.EventMetaProperties
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import java.io.EOFException
+import java.io.File
+import java.io.RandomAccessFile
+import java.nio.channels.FileChannel
+import java.nio.channels.FileLock
+
+private const val FILE_EXTENSION = "stamp"
+
+/**
+ * Controls access to different files used for storing the latest event timestamp of Corda events
+ */
+@Service
+class StateFileStorageControl {
+
+    @Autowired
+    lateinit var eventMetaProperties: EventMetaProperties
+
+    val stateFileStorages: MutableMap<String, StateFileStorage> = mutableMapOf()
+
+    /**
+     * Writes latest timestamp to file
+     * Precaution should be taken since this is a locking call
+     */
+    fun writeTimestamp(eventType: String, timestamp: Long) = getOrCreateStateFileStorage(eventType).writeTimestamp(timestamp)
+
+    /**
+     * Reads latest timestamp from file
+     * Precaution should be taken since this is a locking call
+     */
+    fun readTimestamp(eventType: String): Long = getOrCreateStateFileStorage(eventType).readTimestamp()
+
+    /**
+     * gets correct stateFileStorage or creates a new one if needed
+     */
+    private fun getOrCreateStateFileStorage(eventType: String) : StateFileStorage {
+        synchronized(this) {
```

Comment:
synchronizing on 'this' is generally considered bad practice since you have no control over whoever synchronizes on 'this' outside of this class (risking deadlocks/unnecessary pauses). I'd suggest a field in this class specific for locking.

Something like:

```
val lock = new Object()
```

Even better, you could avoid this altogether by using a ConcurrentMap Kotlin's getOrPut() function: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/java.util.concurrent.-concurrent-map/get-or-put.html